receiver/hostmetricsreceiver/internal/perfcounters/perfcounter_scraper_mock.go

```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package perfcounters
+
+import (
+	"go.opentelemetry.io/collector/internal/processor/filterset"
+)
+
+// MockPerfCounterScraperError returns the supplied errors when Scrape, GetObject,
+// or GetValues are called.
+
+type MockPerfCounterScraperError struct {
```

Comment:
If this is for testing only and no reason to expose it publicly consider to:
Move it to a file that ends with _test.
Use lower case letter to not expose.

Reply:
If the intent is to keep it public add struct and funcs comments :)

Reply:
I do want to keep it public as the intent is for this Mock to be used by several different Scraper packages. Added a bunch of comments